expression/collation.go

```diff
@@ -166,34 +178,23 @@ func DeriveCollationFromExprs(ctx sessionctx.Context, exprs ...Expression) (dstC
 
 		coer := e.Coercibility()
 		ft := e.GetType()
+		collationPriority, ok := collationPriority[strings.ToLower(ft.Collate)]
+		if !ok {
+			collationPriority = -1
```

Comment:
How about returning an error?

Reply:
We can't do that since when collation switch is `off`, many 'fake' collation will be lookup-ed here(not included in `collationPriority`) and we should not return error here.